tests/antithesis/test-template/Dockerfile.client

```diff
@@ -1,35 +1,36 @@
-FROM ubuntu:latest
-
-# Install dependencies
-RUN apt-get update -y && apt-get install -y pip wget
-
-# GO:
-
-# Install go
-RUN wget https://golang.org/dl/go1.22.5.linux-amd64.tar.gz
-RUN tar -C /usr/local -xzf go1.22.5.linux-amd64.tar.gz
-ENV PATH=$PATH:/usr/local/go/bin 
+FROM golang:1.24 AS build
```

Comment:
I prefer the previous approach because this will bring dependabot pull requests, and it may not align with the main branch Go version.

I suggest receiving the Go version as an argument, and when building the Docker image, pass the argument by reading the `.go-version` file.

Something like:

```dockerfile
FROM ubuntu:latest
ARG GO_VERSION
ENV PATH=$PATH:/usr/local/go/bin 

RUN apt-get update -y && apt-get install -y wget && wget "https://golang.org/dl/go${GO_VERSION}.linux-amd64.tar.gz" | tar -C /usr/local xz
```

Then, build with:

```bash
	docker build --build-arg GO_VERSION=$(shell cat $(REPOSITORY_ROOT)/.go-version) -f $(REPOSITORY_ROOT)/tests/antithesis/test-template/Dockerfile.client $(REPOSITORY_ROOT)
```

Reply:
How about instead of using ubuntu and installing, I made it so that the argument is use to pick which official Go docker image to use?

Reply:
That works, too. I'm unsure about providing a default value, as it will be quickly outdated.